openwisp_controller/geo/api/views.py

```diff
@@ -58,4 +87,31 @@ def create_location(self, device):
         return location
 
 
+class GeoJsonLocationList(FilterByOrganizationManaged, generics.ListAPIView):
+    GeoJsonPagination.page_size = 1000
+    permission_classes = (IsAuthenticated,)
+    pagination_class = GeoJsonPagination
+    queryset = Location.objects.filter(devicelocation__isnull=False).annotate(
+        device_count=Count('devicelocation')
+    )
+    serializer_class = GeoJsonLocationSerializer
+
+
+class LocationDeviceList(FilterByParentManaged, generics.ListAPIView):
+    serializer_class = DeviceSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Device.objects.none()
```

Comment:
now that I read the docs, I figured out we're not paginating this. In some cases, a single location (eg: a Hospital) may have hundreds of devices, hence we risk to send too much data and add too much load on the server or cause a timeout.
Can you please add the simple pagination to this endpoint?

Like we have in IPAM: https://github.com/openwisp/openwisp-ipam/blob/master/openwisp_ipam/api/views.py#L53-L56

Reply:
Done :+1: